app/components/tweets/form.rb

```diff
@@ -0,0 +1,40 @@
+module Tweets
+  class Form < ViewComponent::Base
+    include Motion::Component
+    delegate :validation_messages, :valid_class, to: :helpers
+
+    attr_reader :tweet, :on_change
+    map_motion :validate
+    map_motion :save
+
+    def initialize(tweet:, on_change:)
```

Comment:
I'm not familiar with the syntax in the arguments list. Is this the same as doing `initialize({tweet: nil, on_change: nil})`?

Reply:
Not quite.  It specifies that that it will receive two keyword arguments, but does not set a default. So it requires these on initialization.  (You could pass in values of nil, but the caller will still need to set it explicitly.)

Reply:
oh, ok! Thanks for explaining that.